src/qml/filters/fisheye/ui.qml

```diff
@@ -0,0 +1,827 @@
+    function lensValue(index) { return floatFromIndex(index, lensCombo.count) }
+    //scale presets
+    function scalePreset(index) { return floatFromIndex(index, scaleCombo.count) }
+    //scale manual
+    function scaleManual(fValue) { return fValue + 0.5 }
+    //quality presets
+    function qualityIndex(index) { return floatFromIndex(index, qualityCombo.count) }
+    //aspect ratio presets
+    function aspectPreset(index) { return floatFromIndex(index, aspectCombo.count) }
+    //aspect ratio manual
+    function aspectManual(fValue) { return fValue + 0.5 }    
+    //dynamic stretch
+    function dynamicStretch(fValue) { return fValue + 0.5 }
+    //scale Y preset
+    function scaleY(fValue) { return fValue + 0.5 }    
+
+    //calculate a float for plugin. 1-based
+    function floatFromIndex(idx, max) {
+        if (idx < 1)
+            idx = 1
+        if (idx > max)
+            idx = max
+        return ((1 / (max-1)) * (idx-1)).toPrecision(6)
+    }
+
+    //convert float to int for UI inputs. 1-based
+    function indexFromFloat(idx, max) {
+        var v_rng = 1 / max
+        var v_plug = filter.getDouble(idx)
+        for (var i = 1; i < max; i++) {
+            if (v_plug < (v_rng * i)) {
+                return i
+            }
+        }
+        return max
+    }
+
+    //index for reset button. 1-based
+    function setDefaultIndex(value, max){
+        var v_rng = 1 / max
+        for (var i = 1; i < max; i++) {
+            if (value < (v_rng * i))
+                return i
+        }
+        return max
+    }
+
+    //set presets
+    function setPresetData() {
+        var v_more = true
+        var v_ret = ''
+        var i = 0
+
+        do {
+            v_ret = getPresetData(i)
+            if (v_ret !== '' ){
+                setPluginData(v_ret[0], v_ret[1], v_ret[2], v_ret[3], v_ret[4], v_ret[5], v_ret[6], v_ret[7], v_ret[8], v_ret[9], v_ret[10])
+                filter.savePreset(preset.parameters, qsTr(v_ret[iDX_CAMERA]+ ' ' +qsTr(v_ret[iDX_WIDE]) + ' (' +qsTr(v_ret[iDX_RESULT])+')'))
+            } else {
+                v_more = false
+            }
+            i += 1
+        } while(v_more)
+    }
+
+    Component.onCompleted: {
+        filter.blockSignals = true
+        if (filter.isNew) {
+            setPresetData()
+
+            // save (default) in preset dropdown
+            setPluginData(focalRatioDefault, deFishDefault, lensTypeDefault, scalePresetDefault, scaleManualDefault,
+                            qualityDefault, aspectPresetDefault, aspectManualDefault, cropDefault, stretchDefault, scaleYDefault)
+            filter.savePreset(defaultParameters)
+        }
+        filter.blockSignals = false
+
+        if (cameraShowNew) fillCameraCombo()
+
+        setControlData()
+        blockUpdate = false
+    }
+
+
+    GridLayout {
+        columns: 3
+        anchors.fill: parent
+        anchors.margins: 8
+
+        // Row 1 Preset
+        Label {
+            text: qsTr('Presets')

```

Comment:
This might be a British vs. U.S. English thing, but it should be consistent. The U.S. English UI that is the source of all translations including the British uses singular "Preset" and not the plural. This should be consistent.